kafka-connect-bloomberg/src/main/scala/com/datamountaineer/streamreactor/connect/bloomberg/BloombergFieldValueFn.scala

```diff
@@ -84,7 +84,7 @@ private[bloomberg] object BloombergFieldValueFn {
       0
     }
     val offset = ZoneOffset.ofTotalSeconds(offsetSeconds)
-    OffsetTime.of(dt.hour(), dt.minute(), dt.second(), dt.nanosecond(), offset).toString
+    OffsetTime.of(dt.hour()-1, dt.minute(), dt.second(), dt.nanosecond(), offset).toString
```

Comment:
i don't think is right. are you adjusting because of bloomberg works on GMT?  dt should be in GMT, is it not?

Reply:
I agree.  This one didn't seem right with me either, but when running it I noticed that dt.hour() would return 24.  This shouldn't be possible regardless of GMT or not.  There documentation makes no mention of whether it is 0 based or 1 based.
The scala code seems to mirror the jbloomberg utils and you would think they would have seen this issue as well.

Reply:
https://rdrr.io/cran/Rblpapi/man/getTicks.html
`startTime - A Datetime object with the start time, defaults to one hour before current time`

What i'd suggest is to make this -1 as a configurable thing and have it default to -1
There is always the chance that your are getting a conversion between UTC and BST (British Summer Time) - and without me knowing more, maybe on some other regions this scew is different ..?

Reply:
`dt.hour() > 23` can only happen when a field has `dt.hasParts(DateTime.TIME)` set to false - so you shouldn't be using the hour field in that case. Your issue is due to Bloomberg sometimes returning (wrongly IMO) a `Dataype.TIME` datatype although the field has `dt.hasParts(DateTime.TIME)` set to false - in other words the field is flagged as a `TIME` but it's a `DATE`. One way to fix it is to change the logic in `BloombergFieldValueFn` where you would check the presence of the date and time parts regardless of the type (DATE, TIME or DATETIME) and call the appropriate method. BTW a time only field is possible.  
FYI Bloomberg say that it's by design so they won't fix the behaviour.